components/raftstore/src/store/fsm/peer.rs

```diff
@@ -5423,12 +5423,9 @@ where
         fail_point!("on_entry_cache_evict_tick", |_| {});
         if needs_evict_entry_cache(self.ctx.cfg.evict_cache_on_memory_ratio) {
             self.fsm.peer.mut_store().evict_entry_cache(true);
-        }
-        let mut _usage = 0;
-        if memory_usage_reaches_high_water(&mut _usage)
```

Comment:
In the next tick it will call `needs_evict_entry_cache` again, which calls `memory_usage_reaches_high_water` inside. I think checking later again makes more sense than checking right after a compaction.

Also, this further hides the function `memory_usage_reaches_high_water` and keeps the pattern consistent with https://github.com/tikv/tikv/blob/8864e9bac814554bbe4acaddd01282f5fdae3dae/components/raftstore/src/store/fsm/peer.rs#L5350-L5355